### PR TITLE
Add CloudFront invalidation for index files

### LINKED ISSUE #4913: Send invalidations for index.crates.io when updating crates
The HTTP index (index.crates.io) currently has a TTL of 10 minutes and does not send invalidations when publishing new crates. Users may need to wait up to 10 minutes after publishing a crate for it to be visible in the HTTP index.

If we sent invalidations to CloudFront when a crate was updated, we could increase the TTL (for better caching), and users would see updates sooner.

### PR DESCRIPTION
**Pull request by arlosi**
Send invalidation request to CloudFront when a crate is modified.

Does nothing if `CLOUDFRONT_DISTRIBUTION` isn't set.

Fixes #4913

### CODE REVIEW

File: src/worker/cloudfront.rs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        Some(Self {
+            distribution_id,
+            access_key,
+            secret_key,
+        })
+    }
+
+    /// Invalidate a file on CloudFront
+    ///
+    /// `path` is the path to the file to invalidate, such as `config.json`, or `re/ge/regex`
```
Comment (jtgeibel) [inline src/worker/cloudfront.rs:32]: We may want to add a `let path = path.strip_prefix('/').unwrap_or(path);` to handle the case where an initial `/` is included.

**Review by jtgeibel — APPROVED** [review]
Looks great to me, thanks! One small suggestion.

**Comment by arlosi** [comment]
@jtgeibel I've updated the PR to include your suggestion. Can this be merged?

**Comment by jtgeibel** [comment]
Sorry for the delay. I've just merged and will deploy to staging. I've pinged the [infra](https://rust-lang.zulipchat.com/#narrow/stream/242791-t-infra/topic/crates.2Eio.20cloudfront.20environment.20variables/near/302372874) team about getting the environment variables set.

**Pull request merged**